hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -2569,6 +2571,498 @@ private static SessionFactory locateSessionFactoryOnDeserialization(String uuid,
 		throw new InvalidObjectException( "Could not find a SessionFactory [uuid=" + uuid + ",name=" + name + "]" );
 	}
 
+	// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+	// Non-standard HQL functions
+
+	@Override
+	public <T> SqmFunction<T> sql(String pattern, Class<T> type, Expression<?>... arguments) {
+		List<SqmExpression<?>> sqmArguments = new ArrayList<>( expressionList( arguments ) );
+		sqmArguments.add( 0, literal( pattern ) );
+		return getFunctionDescriptor( "sql" ).generateSqmExpression(
+				sqmArguments,
+				getTypeConfiguration().getBasicTypeForJavaType( type ),
+				queryEngine,
+				getJpaMetamodel().getTypeConfiguration()
+		);
+	}
+
+	@Override
+	public SqmFunction<String> format(Expression<? extends TemporalAccessor> datetime, String pattern) {
+		SqmExpression<String> patternLiteral = value( pattern );
```

Comment:
I actually believe that this must be a literal:

```suggestion
		SqmExpression<String> patternLiteral = literal( pattern );
```

Reply:
Actually it had to be a `SqmFormat` to allow for dialect translation of the pattern, now it's fixed.